consensus/istanbul/backend/engine.go

```diff
@@ -285,6 +286,57 @@ func (sb *backend) VerifyHeaders(chain consensus.ChainReader, headers []*types.H
 	return abort, results
 }
 
+func (sb *backend) disconnectReleasedVal(header *types.Header, curVals istanbul.ValidatorSet) error {
+	if curVals == nil {
+		return errors.New("validator set is nil")
+	}
+	if header == nil {
+		return errors.New("header is nil")
+	}
+	// Istanbul BFT chain does not have nil for `sb.broadcaster`,
+	// This addition is for existing testbase that did not correctly intiialize
+	// braodcaster. Theferoe, it can be interpreted that the network setup
+	// is not correctly configured, which is not the case of production
+	if sb.broadcaster == nil {
+		return nil
+	}
+
+	var ppNumber uint64
+	switch header.Number.Uint64() {
+	case 0, 1:
```

Comment:
I think the below looks more readable.
```
ppNumber := uint64(0)
if header.Number.Uint64() > 1 {
    ppNumber = header.Number.Uint64() - 2
}
```
If not, I think `case 1:` is enough since `header.Number.Uint64() == 0` is infeasible.